apis/Google.Datastore.V1Beta3/Google.Datastore.V1Beta3.Snippets/DatastoreDbSnippets.cs

```diff
@@ -90,10 +91,11 @@ public void StructuredQuery()
             {
                 Filter = Filter.Equal("author", "Jane Austen")
             };
-            DatastoreQueryResults results = db.RunQuery(query);
-            // DatastoreQueryResults implements IEnumerable<Entity>, but you can
-            // call AsEntityResults(), AsBatches() or AsResponses() to see the query
-            // results in whatever way makes most sense for your application.
+            LazyDatastoreQuery results = db.RunQuery(query);
```

Comment:
The reason for keeping RunQuery as the method name was to connect with the RPC API - any users who look at the RPCs (or other platforms exposing the RPCs directly) is likely to look for RunQuery.

I wouldn't mind an API on DatastoreDb of:

```csharp
DatastoreQueryResults RunQueryEagerly(...)
LazyDatastoreQuery RunQueryLazily(...)
```

... and we could change LazyDatastoreQuery to make *that* the point at which you decide sync/async, as well as the view you want:

```csharp
IEnumerable<Entity> GetEntities()
IAsyncEnumerable<Entity> GetEntitiesAsync()
IEnumerable<QueryResponse> GetRpcResponses()
IAsyncEnumerable<QueryResponse> GetRpcResponsesAsync()
```

That means you'd always need to call one of them, e.g.

```csharp
var lazyQuery = db.RunQueryLazily(...);
foreach (var entity in lazyQuery.GetEntities())
{
   ...
}
```

That's slightly ugly in some ways, but maybe it's appropriate...